edb/schema/objects.py

```diff
@@ -71,24 +89,30 @@ def default_field_merge(target: Object, sources: List[Object],
 
 
 class ComparisonContextWrapper:
-    def __init__(self, context, pair):
+    def __init__(self, context: ComparisonContext, pair: Pair) -> None:
         self.context = context
         self.pair = pair
 
-    def __enter__(self):
+    def __enter__(self) -> None:
         self.context.push(self.pair)
 
-    def __exit__(self, exc_type, exc_value, traceback):
+    def __exit__(
+        self, exc_type: type, exc_value: Exception, traceback: Any
```

Comment:
Strictly speaking, `exc_type`, `exc_value` and `traceback` are optional in `__exit__`.  It's a bit annoying that Mypy forces you to explicitly type methods that it should know the signature of, them being part of the language spec.

Reply:
I'll fix the optional bit.

Reply:
`traceback: Any` -> `traceback: Optional[types.TracebackType]`
`exc_type: type` -> `exc_type: Optional[Type[Exception]]`